tests/unit/rest/test_client.py

```diff
@@ -44,3 +50,102 @@ def test_obsolete_exception_twiliotaskrouterclient(self):
     def test_obsolete_exception_twiliotrunkingclient(self):
         self.assertRaises(ObsoleteException, TwilioTrunkingClient,
                           "Expected raised ObsoleteException")
+
+
+class TestRegionEdgeClients(unittest.TestCase):
+    def setUp(self):
+        self.http_client_patcher = patch('twilio.rest.TwilioHttpClient')
+
+        self.http_client_mock = Mock(wraps=TwilioHttpClient())
+        self.request_mock = Mock()
+
+        self.http_client_mock.request.return_value = Response(500, '')
```

Comment:
Test cases look good but all the HTTP mocking could be dropped if the hostname logic was moved to its own function and explicitly tested.